### PR TITLE
Add ingressgateway_tls config and changes to the benchmark runner

### PR DESCRIPTION
**Pull request by VillePihlava**
This pull request adds the possibility to use the Fortio client in the benchmark runner with https to connect to the ingress gateway. Changes to the benchmark runner have been added  to support this change. The changes also allow for changing the directory where the benchmark runner finds configurations for custom benchmark configurations. Also an example ingressgateway_tls benchmark config has been created.

Changes include:
 - Adding Fortio flags to the benchmark runner
	- add uniform flag
	- add nocatchup flag
	- add keepalive flag
	- add connection-reuse flag
 - Enabling custom benchmark config directory
 - Enabling setting specific Istio version
 - Changes to the helm template for using https
 - Creating self signed certificate and using it with https
 - Set perf_record to false until flame graphs are confirmed to work

Contributes to: https://github.com/istio/istio/issues/41609

### CODE REVIEW

**Comment by VillePihlava** [comment]
Some additional thoughts on the contents of this pull request:
* A possibility to configure the certificate for testing different types could be added in the future
* Fortio is now configured to use the -cacert flag but the -https-insecure flag could probably be used as well
* I'm not familiar with how the port exclusion in Istio works so it is unclear to me whether it is necessary in this case similarly to how it is used here:
https://github.com/istio/tools/blob/b05f8cb30f1db5ae88b25ebac399671682bd6f24/perf/benchmark/templates/fortio.yaml#L373-L374

File: perf/benchmark/templates/fortio.yaml [hunk 1]
```diff
@@ -394,6 +394,11 @@ spec:
       volumes:
       - name: shared-data
         emptyDir: {}
+      {{- if and (eq $.name "fortioclient") ($.Values.cert.server) }}
+      - name: gateway-secret
+        secret:
+          secretName: fortio-server-ingress-cert
```
Comment (howardjohn) [inline perf/benchmark/templates/fortio.yaml:400]: Don't mount, use credentialName in `Gateway`. Mounting is the legacy way

Comment (VillePihlava) [reply perf/benchmark/templates/fortio.yaml:400]: If I understand your comment correctly, I think I'm already doing that here:
https://github.com/istio/tools/blob/f1c4203fe9fc8382e0fb8eedd1af591026b2b344/perf/benchmark/templates/fortio.yaml#L602-L612

I mount the secret to use it with Fortio's `-cacert` flag. The parameter for Fortio is given here:
https://github.com/istio/tools/blob/f1c4203fe9fc8382e0fb8eedd1af591026b2b344/perf/benchmark/configs/istio/ingressgateway_tls/cpu_mem.yaml#L25

I left an old configuration which I think uses the mount-based approach with a comment, should I just remove it?
https://github.com/istio/tools/blob/f1c4203fe9fc8382e0fb8eedd1af591026b2b344/perf/benchmark/templates/fortio.yaml#L589-L601

Comment (howardjohn) [reply perf/benchmark/templates/fortio.yaml:400]: ohh sorry, misunderstood this was for fortio. Fine to mount for fortio, for all istio things we should use credentialName

Comment (luukasmakila) [reply perf/benchmark/templates/fortio.yaml:400]: Is this good to go? @howardjohn

**Review by howardjohn — APPROVED** [review]

**Pull request merged**